### ISSUE TITLE
glTF 2.0 model fails to compress with Draco compression

### ISSUE THREAD

**Issue by shehzan10** [AnalyticalGraphicsInc/gltf-pipeline#380]
The attached glTF 2.0 model fails with draco compression on with `draco.compressionLevel >= 5`. The conversion call returns an encoded length of 0.

[model_with_textures.gltf.txt](https://github.com/AnalyticalGraphicsInc/gltf-pipeline/files/2120680/model_with_textures.gltf.txt)

@FrankGalligan is this something you could look into?

**Comment by FrankGalligan** [comment]
Hi @shehzan10,

So I was able to look at this gltf file. The issue is the second and third mesh are both made up of one primitive. Each primitive is made up of two triangles and all the triangles are degenerate. 

In the case draco.compressionLevel <= 4 we are stripping the duplicate points.

Anyway I will have to think a little bit about what to do in these situations. Are the degenerate triangles needed?

**Comment by FrankGalligan** [comment]
I looked into this more. This is basically working as intended. We could always fallback to a lower compression level if the higher compression level returns an error (Kind of like what we do with morph targets). But in this situation the data would still not be usable, so I don't know if the falling back is the best idea.

What do you think?

**Comment by shehzan10** [comment]
> So I was able to look at this gltf file. The issue is the second and third mesh are both made up of one primitive. Each primitive is made up of two triangles and all the triangles are degenerate.

Thanks for looking at this. It's interesting what you mentioned. I'm fairly sure I have steps before generating the glTF to remove degenerate triangles. So let me look at what's going on here and if I can fix it on my end, then I think that's the best route to take.

**Comment by shehzan10** [comment]
> So I was able to look at this gltf file. The issue is the second and third mesh are both made up of one primitive. Each primitive is made up of two triangles and all the triangles are degenerate.

@FrankGalligan I'm just trying to understand the error better here. Under what circumstances is this error arising? As in, is it because all triangles within all the primitives of a mesh are degenerate?

I started with the model I uploaded here, removed the degenerate primitives and the draco compression worked fine. So far so good.

I was able to isolate the sub-model in which the degenerate triangles existed (circled in the image below).
[sub-model-no-textures.gltf2.txt](https://github.com/AnalyticalGraphicsInc/gltf-pipeline/files/2229397/sub-model-no-textures.gltf2.txt)

![image](https://user-images.githubusercontent.com/2288659/43224942-d5bce98e-9025-11e8-8d88-693c66cb5b6d.png)

This model has all the same degenerate triangle geometry/mesh/primitives etc including the degenerate primitives (different values though). However, when I run this, draco compresses this just fine. The only difference here is the textures have been removed, so the texture coordinates are no longer used.

So if Draco was erroring out for degenerate triangles, shouldn't it error out for this model as well?

Ps. This is not to say that degenerate triangles are good. I'm adding extra checks to remove them from my pipeline.
